src/gui/uithememanager.cpp

```diff
@@ -133,7 +133,15 @@ QIcon UIThemeManager::getIcon(const QString &iconId, const QString &fallback) co
 QIcon UIThemeManager::getFlagIcon(const QString &countryIsoCode) const
 {
     if (countryIsoCode.isEmpty()) return {};
-    return QIcon(":/icons/flags/" + countryIsoCode.toLower() + ".svg");
+
+    static QHash<QString, QIcon> flagCache;
```

Comment:
Despite the fact that it's used in existing code it's incorrect to have static variables for caches. We should use (mutable) class fields instead.